packages/babel-core/src/config/full.ts

```diff
@@ -250,10 +252,10 @@ const makeDescriptorLoader = <Context, API>(
 
     const externalDependencies: Array<string> = [];
 
-    let item = value;
+    let item: unknown = value;
     if (typeof value === "function") {
       const factory = maybeAsync(
-        value,
+        value as (api: API, options: {}, dirname: string) => unknown,
```

Comment:
I wish I could get rid of the typing cast here, the typeof check here can not rule out `{}` completely as it could be `Function` object.

Reply:
@ehoogeveen-medweb My bad. I was meant for `object` as I typed `value` as an union of `object` and a specific function:
https://github.com/babel/babel/blob/0e20ec5e3dbc4aeae042c1468bb772a6f4ab6a01/packages/babel-core/src/config/config-descriptors.ts#L38

but for some reason the `typeof` check does not rule out the `object` completely.